### PR TITLE
EPMLABSBRN-95 update endpoint to get tasks

### PR DESCRIPTION
**Pull request by goopnigoop**
- added logic to Controller -> Service -> Repository levels
- test for levels are listed above
- fixed issues and improvements
- to avoid 'Unchecked cast:' warnings added @Suppress("UNCHECKED_CAST")
  for special cases
- added kotlin-test for testing exception cases using assertFailsWith

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/controller/TaskController.kt [hunk 1]
```diff
@@ -21,19 +24,32 @@ class TaskController(private val taskService: TaskService, private val uploadTas
     private val log = logger()
 
     @GetMapping
-    fun getTasksForExerciseId(@RequestParam(value = "exerciseId", required = false) exerciseId: Long?): List<TaskDto> {
-        // TODO will be improve, search will be more flexible
-
+    @ApiOperation("Get all tasks")
+    fun getTasksByExerciseId(
+        @RequestParam(value = "exerciseId", required = false) exerciseId: Long?
```
[the remark this answers is not in the document]
Comment (goopnigoop) [reply src/main/kotlin/com/epam/brn/controller/TaskController.kt]: we can get tasks without exercise id, I'll rename method

**Comment by goopnigoop** [comment]
> could you please add TasksControllerIT

ok

File: src/main/kotlin/com/epam/brn/controller/TaskController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/controller/TaskController.kt]: in which cases we need to get tasks not by exId?

[the remark this answers is not in the document]
Comment (goopnigoop) [reply src/main/kotlin/com/epam/brn/controller/TaskController.kt]: we don't?

**Pull request merged**